docs/reference/judoc.md

```diff
@@ -1 +1,93 @@
-# Judoc reference
+# Judoc syntax reference
+Judoc is used to document parts of your code. You can attach *Judoc
+blocks* to the following entities:
+
+1. A module.
+2. A type definition.
+4. A constructor definition.
+3. A type signature of a function.
+4. An axiom definition.
+
+In order to attach documentation to either of these items, write *blocks* of
+documentation before them:
+
+1. For modules:
+   ```
+   --- This module is cool
+   module Cool;
+   ..
+   ```
+2. For type definitions:
+   ```
+   --- Unary representation of natural numbers
+   type Nat : Type :=
+     | --- Nullary constructor representing number 0
+       zero : Nat;
+     | --- Unary constructor representing the successor of a natural number
+       suc : Nat -> Nat;
+   ```
+3. For type signatures (and likewise for axioms):
+   ```
+   --- The polymorphic identity function
+   id : {A : Type} -> A -> A;
+   ```
+   
+   Next we define the syntax of Judoc *blocks*.
+
+## Block
+
+A *block* can be one of these:
+1. A *paragraph*.
+2. An *example*.
+
+*Blocks* are separated by a line with only `---`.
+For instance, this is a sequence of two *blocks*:
+```
+--- First block
+---
+--- Second block
+```
+
+Note that the following is a single block since it lacks the `---` separator:
+```
+--- First block
+
+
+--- Still first block
```

Comment:
Suggestion: Have 1 blank line between the two block to make it clear that only 1 and not 2 blank lines are required.

```suggestion
--- First block

--- Still first block
```